### PR TITLE
Mac bug fixes

### PR DESCRIPTION
**Pull request by timrisi**

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build success

File: src/Foundation/NSUrl.cs [hunk 1]
```diff
@@ -142,6 +142,19 @@ public bool SetResource (NSString nsUrlResourceKey, NSObject value)
 			return SetResourceValue (value, nsUrlResourceKey, out error);
 		}
 
+		public int Port
```
Comment (rolfbjarne) [inline src/Foundation/NSUrl.cs]: style: opening brace goes on the same line as the property (and for the same line as the `get` keyword as well)

File: src/AppKit/NSImage.cs [hunk 2]
```diff
@@ -193,6 +194,8 @@ public static NSImage ImageNamed (NSImageName name)
 				return ImageNamed (NSImageNameUserGuest);
 			case NSImageName.MobileMe:
 				return ImageNamed (NSImageNameMobileMe);
+			case NSImageName.ShareTemplate:
+				return ImageNamed(NSImageNameShareTemplate);
```
Comment (mandel-macaque) [inline src/AppKit/NSImage.cs]: Style, missing space after ImageNamed.

File: src/Foundation/NSUrl.cs [hunk 3]
```diff
@@ -142,6 +142,19 @@ public bool SetResource (NSString nsUrlResourceKey, NSObject value)
...
+		{
+			get
+			{
+				var number = this.PortNumber;
+
+				if (number == null)
+					return -1;
+
+				return (int)number;
```
Comment (mandel-macaque) [inline src/Foundation/NSUrl.cs]: I believe you can re-write the code as:
```csharp
    get 
    {
        return (this.PortNumber == null) ? -1: (int) this.PortNumber;
    }
```
If getting the PortNumber is an expensive operation you can use a temp var and do the same unary operation.

Comment (KanLei) [inline src/Foundation/NSUrl.cs]: How about this `return (int)(this.PortNumber ?? -1);`

**Comment by xamarin-release-manager** [comment]
Build success

File: src/Foundation/NSUrl.cs [hunk 4]
```diff
@@ -142,6 +142,12 @@ public bool SetResource (NSString nsUrlResourceKey, NSObject value)
 			return SetResourceValue (value, nsUrlResourceKey, out error);
 		}
 
+		public int Port {
+			get {
+				return (int) (this.PortNumber ?? -1);
+			}
```
Comment (mandel-macaque) [inline src/Foundation/NSUrl.cs]: Much nicer!!!! I did not think of that way, love it!

**Comment by mandel-macaque** [comment]
LGTM

**Comment by spouliot** [comment]
LGTM @chamons ?

**Comment by chamons** [comment]
LGTM.

**Pull request merged**